lightwood/api/data_source.py

```diff
@@ -198,6 +200,8 @@ def __getitem__(self, idx):
                         custom_data = {col_name: [None]}
 
                     sample[feature_set][col_name] = self.get_encoded_column_data(col_name, custom_data=custom_data)[0]
+                elif 'previous_' in col_name:   # TODO: fragile check for name
```

Comment:
See comments above on this check (note, if that one is too hard checking with `.startswith` is still better.